runtime/src/slots.rs

```diff
@@ -0,0 +1,1425 @@
+		// accordingly.
+		for (maybe_new_deploy, para_id, amount, range) in winners.into_iter() {
+			match maybe_new_deploy {
+				Some(bidder) => {
+					// For new deployments we ensure the full amount is deducted. This should always
+					// succeed as we just unreserved the same amount above.
+					if T::Currency::withdraw(
+						&bidder.who,
+						amount,
+						WithdrawReason::Fee,
+						ExistenceRequirement::AllowDeath
+					).is_err() {
+						continue;
+					}
+
+					// Add para IDs of any chains that will be newly deployed to our set of managed
+					// IDs
+					<ManagedIds<T>>::mutate(|m| m.push(para_id));
+
+					Self::deposit_event(RawEvent::WonDeploy(bidder.clone(), range, para_id, amount));
+
+					// Add a deployment record so we know to on-board them at the appropriate
+					// juncture.
+					let begin_offset = <LeasePeriodOf<T>>::sa(range.as_pair().0 as u64);
+					let begin_lease_period = auction_lease_period_index + begin_offset;
+					<OnboardQueue<T>>::mutate(begin_lease_period, |starts| starts.push(para_id));
+					let entry = (begin_lease_period, IncomingParachain::Unset(bidder));
+					<Onboarding<T>>::insert(&para_id, entry);
+				}
+				None => {
+					// For renewals, reserve any extra on top of what we already have held
+					// on deposit for their chain.
+					let extra = if let Some(additional) =
+						amount.checked_sub(&Self::deposit_held(&para_id))
+					{
+						if T::Currency::withdraw(
+							&para_id.into_account(),
+							additional,
+							WithdrawReason::Fee,
+							ExistenceRequirement::AllowDeath
+						).is_err() {
+							continue;
+						}
+						additional
+					} else {
+						Default::default()
+					};
+					Self::deposit_event(RawEvent::WonRenewal(para_id, range, amount, extra));
+				}
+			}
+
+			// Finally, we update the deposit held so it is `amount` for the new lease period
+			// indices that were won in the auction.
+			let current: u64 = lease_period_index.as_();
+			let current = current as usize;
+			let index: u64 = auction_lease_period_index.as_();
+			let index = index as usize;
+
+			if let Some(offset) = index.checked_sub(current) {
+				// Should always succeed; for it to fail it would mean we auctioned a lease period
+				// that already ended.
+
+				// The lease period index range (begin, end) that newly belongs to this parachain
+				// ID. We need to ensure that it features in `Deposits` to prevent it from being
+				// reaped too early (any managed parachain whose `Deposits` set runs low will be
+				// removed).
+				let pair = range.as_pair();
+				let pair = (pair.0 as usize + offset, pair.1 as usize + offset + 1);
+				<Deposits<T>>::mutate(para_id, |d| {
+					// Left-pad with zeroes as necessary.
+					if d.len() < pair.0 {
+						d.resize_with(pair.0, Default::default);
+					}
+					// Then place the deposit values for as long as the chain should exist.
+					for i in pair.0 .. pair.1 {
```

Comment:
It looks like you increment `pair.1` by one above in order to make this for loop go over all the indices you want.

I think it is more clear if you do not increment `pair.1` above, but write the loop like `pair.0 ..= pair.1`

If I am misunderstanding, let me know.